src/components/ChartControls.jsx

```diff
@@ -10,9 +10,11 @@ import DrawTools from './DrawTools.jsx';
 import ChartSetting from './ChartSetting.jsx';
 import Share from './Share.jsx';
 
-const ChartControls = () => (
+const ChartControls = ({
+    isMobile
+}) => (
     <div className="cq-chart-controls">
-        <CrosshairToggle />
+        {isMobile ? '' : <CrosshairToggle />}
```

Comment:
Just make `isMobile` an `@observable`, and have its value set by props. You can then access the value via `this.mainStore.chart.isMobile`. When `isMobile` is changed, all `jsx` files using it will be updated as well.